src/realm/sort_descriptor.cpp

```diff
@@ -549,14 +556,28 @@ DescriptorOrdering& DescriptorOrdering::operator=(const DescriptorOrdering& rhs)
     return *this;
 }
 
-void DescriptorOrdering::append_sort(SortDescriptor sort)
```

Comment:
I know I suggested having `append_sort` and `prepend_sort` API, but looking at how similar the implementation of these methods is, would it make sense to introduce a `MergeMode` enum with the values `append` and `prepend`? Then we can have only one method - `add_sort` which will be used like `ordering.add_sort(sort_descriptor, MergeMode::append)`.

And I guess to avoid making it a breaking change, you can keep `append_sort` that calls `add_sort(sort, MergeMode::prepend)` and mark it as deprecated (not sure if that's a concern in Core).

Disclaimer: I don't have much insight into Core's codebase, so if my suggestion wouldn't be idiomatic, I'm totally fine with keeping things as they are - the current state of the API would work just fine for .NET.

Reply:
Sure, that's a good idea; it's more extensible, and reduces some copied code. I don't think we should deprecate anything at this point though, I don't want to break the existing behaviour on table views in case people are relying on it.

Reply:
By deprecate, I meant annotate it as a deprecated attribute to tell consumers about the new API without breaking their builds/runtime behavior. Something like:

```cpp
[[deprecated("use add_sort(sort, MergeMode::prepend) instead")]]
void DescriptorOrdering::append_sort(SortDescriptor sort)
{
    add_sort(sort, MergeMode::prepend);
}

void DescriptorOrdering::add_sort(SortDescriptor sort, MergeMode mode)
{
    // actual implementation
}
```

Reply:
Oh ok, I thought you were wanting to deprecate the prepend behaviour, but I see you are referring to the naming. I opted to keep the existing method name and add `MergeMode::prepend` as the default parameter so that there is no breaking change and SDKs can update without changing any code, while still providing the new merge option if explicitly requested. Regarding naming, I've gone back and forth myself, but I think "append" is still appropriate here, because the context is `DescriptorOrdering` and the sort descriptors are indeed always appended with regards to other descriptor types (limit, distinct). I think having to write out `SortDescriptor::MergeMode::<append/prepend>` should be clear enough to denote what the parameter applies to (but I'm a bit biased so if you strongly disagree let me know).